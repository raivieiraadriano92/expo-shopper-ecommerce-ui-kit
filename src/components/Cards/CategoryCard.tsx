import { FunctionComponent } from 'react'

import { HStack, Image, Text } from 'native-base'
import { TouchableOpacity } from 'react-native'

type Props = {
  imageUri: string
  onPress(): void
  title: string
  totalProducts?: number
}

export const CategoryCard: FunctionComponent<Props> = ({
  imageUri,
  onPress,
  title,
  totalProducts
}) => (
  <TouchableOpacity {...{ onPress }}>
    <HStack
      alignItems="center"
      bg="gray.50"
      borderRadius="2xl"
      h={22}
      justifyContent="space-between"
      px={3}>
      <Text variant="body2">{title}</Text>
      <HStack alignItems="center" space={4}>
        {!!totalProducts && (
          <Text variant="body1">{`${totalProducts} product${totalProducts > 1 ? 's' : ''}`}</Text>
        )}
        <Image
          alt="Alternate Text"
          borderRadius="xl"
          resizeMode="contain"
          size={13}
          source={{
            uri: imageUri
          }}
        />
      </HStack>
    </HStack>
  </TouchableOpacity>
)
