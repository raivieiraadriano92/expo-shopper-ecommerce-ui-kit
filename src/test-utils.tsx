import { FunctionComponent } from 'react'

import { render } from '@testing-library/react-native'

import { AppWrapper } from '../src/AppWrapper'

const Wrapper: FunctionComponent = ({ children }) => <AppWrapper>{children}</AppWrapper>

const customRender: typeof render = (ui, options) => render(ui, { wrapper: Wrapper, ...options })

// re-export everything
export * from '@testing-library/react-native'

// override render method
export { customRender as render }
