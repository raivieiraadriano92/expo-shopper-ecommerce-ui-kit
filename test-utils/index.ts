import { render } from '@testing-library/react-native'
import { AppWrapper } from 'src/AppWrapper'

const customRender: typeof render = (ui, options) => render(ui, { wrapper: AppWrapper, ...options })

// re-export everything
export * from '@testing-library/react-native'

// override render method
export { customRender as render }
