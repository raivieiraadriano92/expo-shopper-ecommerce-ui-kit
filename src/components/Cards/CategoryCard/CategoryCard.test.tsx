import { render, waitFor } from 'test-utils'

import { CategoryCard } from './CategoryCard'

describe('<CategoryCard />', () => {
  it('renders correctly', async () => {
    //@ts-ignore
    const { findByText, toJSON } = render(
      <CategoryCard
        imageUri="https://philco.vteximg.com.br/arquivos/ids/175970/PFO01BTP.jpg?v=637111522912700000"
        onPress={() => {}}
        title="category title"
      />
    )

    await waitFor(() => findByText('category title'))

    expect(toJSON()).toMatchSnapshot()
  })
})
