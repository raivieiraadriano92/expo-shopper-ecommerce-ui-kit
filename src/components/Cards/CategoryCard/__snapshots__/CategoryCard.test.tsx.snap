// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryCard /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F4F5F6",
          "borderRadius": 16,
          "flexDirection": "row",
          "height": 88,
          "justifyContent": "space-between",
          "paddingLeft": 12,
          "paddingRight": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#1f2937",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 14,
            "fontStyle": undefined,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 22,
            "textDecorationLine": undefined,
          }
        }
      >
        category title
      </Text>
      <View
        style={Object {}}
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Image
          accessibilityLabel="Alternate Text"
          alt="Alternate Text"
          onError={[Function]}
          resizeMode="contain"
          source={
            Object {
              "uri": "https://philco.vteximg.com.br/arquivos/ids/175970/PFO01BTP.jpg?v=637111522912700000",
            }
          }
          style={
            Object {
              "borderRadius": 12,
              "height": 52,
              "maxWidth": "100%",
              "width": 52,
            }
          }
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
